File: Data Structures & Algorithms/maximum-product-subarray/submission-2.cpp
class Solution {
public:
    int maxProduct(vector<int>& nums) {
        int minProd = nums[0];
        int maxProd = nums[0];
        int result = nums[0];

        for(int i = 1;i < nums.size();i++){
            int curr = nums[i];
            if(curr < 0){
                swap(minProd,maxProd);
            }
           
            maxProd = max(curr,curr*maxProd);
            minProd = min(curr,curr*minProd);

            result = max(result,maxProd);
        }
        return result;
    }
};

File: Data Structures & Algorithms/house-robber/submission-2.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        dp = [-1] * len(nums)
        dp[0] = nums[0]

        for i in range(1,len(nums)):
            if i == 1:
                dp[i] = max(dp[0],nums[i])
            else:
                dp[i] = max(dp[i - 1],nums[i] + dp[i - 2])
        
        return dp[len(nums) - 1]